test/Microsoft.AspNetCore.Razor.Test.MvcShim/Microsoft.AspNetCore.Mvc.ApplicationParts/ProvideApplicationPartFactoryAttribute.cs

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Mvc.ApplicationParts
+{
+    [AttributeUsage(AttributeTargets.Assembly, AllowMultiple = false)]
+    public sealed class ProvideApplicationPartFactoryAttribute : Attribute
+    {
+        public ProvideApplicationPartFactoryAttribute(Type factoryType)
+        {
+        }
+
+        public ProvideApplicationPartFactoryAttribute(string factoryTypeName)
```

Comment:
Unfortunately, `WriteCodeFragment` only supports writing string attributes. I feel this is a better alternative to writing out a file using `WriteLinesToFile`